File: benchmark-java-sync/src/test/java/com/example/benchmarkjavasync/core/ApplicationServiceImplTest.java
package com.example.benchmarkjavasync.core;

import com.example.benchmarkjavasync.core.model.ApplicationServiceRequest;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;

import java.math.BigDecimal;
import java.util.Optional;
import java.util.UUID;

@SpringBootTest
class ApplicationServiceTest {

    @Autowired
    ApplicationService<Optional<UUID>> applicationService;

    @Test
    void test(){
        applicationService.process(new ApplicationServiceRequest(
                "NAme",
                "LastName",
                1234,
                123456,
                BigDecimal.TEN,
                "New"
        ));
    }
}

File: benchmark-kotlin-async/src/test/kotlin/com/example/benchmarkkotlinasync/impl/ApplicationServiceImplTest.kt
package com.example.benchmarkkotlinasync.impl

import com.example.benchmarkkotlinasync.core.model.ApplicationServiceRequest
import kotlinx.coroutines.runBlocking
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import java.math.BigDecimal

@SpringBootTest
class ApplicationServiceImplTest {

    @Autowired
    lateinit var sut: ApplicationServiceImpl

    @Test
    fun process() {
        runBlocking {
            val result = sut.process(
                ApplicationServiceRequest(
                    "Test",
                    "TestLN",
                    1234,
                    123456,
                    BigDecimal.TEN,
                    "!"
                )
            )

            println(result)
        }
    }
}